collectors/health.go

```diff
@@ -1206,76 +1206,40 @@ func (c *ClusterHealthCollector) collect(ch chan<- prometheus.Metric) error {
 			"forced_backfill": &forcedBackfillPGs,
 			"down":            &downPGs,
 		}
+		pgStateGaugeMap = map[string]*prometheus.Gauge{
+			"degraded":        &c.DegradedPGs,
+			"active":          &c.ActivePGs,
+			"unclean":         &c.UncleanPGs,
+			"undersized":      &s.UndersizedPGs,
+			"peering":         &c.PeeringPGs,
+			"stale":           &c.StalePGs,
+			"scrubbing":       &c.ScrubbingPGs,
+			"scrubbing+deep":  &c.DeepScrubbingPGs,
+			"recovering":      &c.RecoveringPGs,
+			"recovery_wait":   &c.RecoveryWaitPGs,
+			"backfilling":     &c.BackfillingPGs,
+			"backfill_wait":   &c.BackfillWaitPGs,
+			"forced_recovery": &c.ForcedRecoveryPGs,
+			"forced_backfill": &c.ForcedBackfillPGs,
+			"down":            &c.DownPGs,
+		}
 	)
 
 	for _, p := range stats.PGMap.PGsByState {
-		for pgState := range pgStateMap {
+		for pgState := range pgStateCounterMap {
 			if strings.Contains(p.States, pgState) {
-				*pgStateMap[pgState] += p.Count
+				*pgStateCounterMap[pgState] += p.Count
 			}
 		}
 	}
 
-	if *pgStateMap["degraded"] > 0 {
```

Comment:
It makes me so happy that you got rid of this line of `if` statements!

Reply:
A combination of golang and the prom library actually makes this inordinately difficult, since `prometheus.Gauge` is actually an interface, and you can't have pointers to an interface type, so back to the drawing board...

Reply:
Derp, I got confused by some other test output when I removed the `&` from the map literal. I suppose it works because `c` is a pointer receiver. I'm always a little surprised by how nicely references work in golang.